toolchains/cc/cc.bzl

```diff
@@ -123,7 +123,7 @@ def _parse_cc_toolchain_info(content, filename):
 def _nixpkgs_cc_toolchain_config_impl(repository_ctx):
     host_cpu = get_cpu_value(repository_ctx)
     cross_cpu = repository_ctx.attr.cross_cpu or host_cpu
-    darwin = host_cpu == "darwin" or host_cpu == "darwin_arm64"
+    darwin = (host_cpu == "darwin" or host_cpu == "darwin_arm64") and cross_cpu == host_cpu
```

Comment:
Before this change, when cross-compiling from Linux to MacOS, `darwin` would be `True` and the `target_libc` would be `"macosx"`, but after this change `darwin` is `False`, so `target_libc` is `"local"`. Is that correct/intentional? @benradf @DolceTriade

Reply:
Good catch!

Is this bit correct though:
> Before this change, when cross-compiling from Linux to MacOS, `darwin` would be `True`

I could well be mistaken, but in this case [wouldn't](https://github.com/bazelbuild/bazel/blob/b27ca73617c5319d4b15a9e5ba499cf476679a95/tools/cpp/lib_cc_configure.bzl#L209) `host_cpu` be `k8`, and so `darwin` would be `False`?

Either way, this needs fixing. I'll raise a pull request and we can discuss the details there.

Reply:
[Fix darwin flag when cross-compiling from Linux to MacOS #413 ](https://github.com/tweag/rules_nixpkgs/pull/413)

Reply:
> I could well be mistaken, but in this case [wouldn't](https://github.com/bazelbuild/bazel/blob/b27ca73617c5319d4b15a9e5ba499cf476679a95/tools/cpp/lib_cc_configure.bzl#L209) host_cpu be k8, and so darwin would be False?

Yes, I think you are correct; I was reading that backwards, but it did look suspicious regardless! :)